manifests/extension.pp

```diff
@@ -66,6 +70,7 @@
   $settings          = {},
   $settings_prefix   = false,
   $sapi              = 'ALL',
+  $pipe              = undef,
```

Comment:
Does `$pipe` then not also have to be passed as an argument to `ensure_packages`?